### DISCUSSION TITLE
brew service segfaults on Apple Silicon

### DISCUSSION THREAD

**Discussion by thsubaku9** [rabbitmq/rabbitmq-server#3129] [Other]
RabbitMQ doesn't seem to be working on apple silicon (tried using rosetta and installed using brew). Reproduction steps as follows ->

1.` arch -x86_64 brew install rabbitmq`
2. start the brew service-> local socket address is not accessible
3. alternatively don't start the brew services and run `rabbitmq-server` 
/usr/local/sbin/rabbitmq-server: line 153: 51226 Segmentation fault: 11  start_rabbitmq_server "$@"

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
RabbitMQ has no CPU architecture-specific parts. As long as a [compatible version of Erlang](https://www.rabbitmq.com/which-erlang.html) is installed, you are good to go. Erlang 24 definitely runs on ARM-based Macs.

If you use an ARM-based machine, I'm not sure what the `arch -x86_64` is for. Otherwise the only relevant bit of information here
is that the process segfaults. Again, RabbitMQ does not allocate memory directly so what segfaults here is Erlang/OTP, possibly
because `arch -x86_64` has instructed Homebrew to target the wrong architecture?

**Comment by michaelklishin** [comment]
According to [this Hombrew formula page](https://formulae.brew.sh/formula/erlang), `erlang@24` supports Apple Silicon on Big Sur.
Perhaps you should stop using Rosetta, uninstall both RabbitMQ and Erlang formulae and reinstall them without Rosetta.

Alternatively Erlang 24 can be [installed using `kerl`](https://www.rabbitmq.com/which-erlang.html#erlang-repositories) which builds
it from source. It should not be necessary for most users but is always an option.

RabbitMQ supports Erlang 24 as of [`3.8.16`](https://www.rabbitmq.com/which-erlang.html), and the `rabbitmq` formula [already provides `3.8.17`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/rabbitmq.rb) so you should be able to
install `erlang@24` and then `rabbitmq`.

We do not maintain neither formulae, so please direct all questions to their maintainers.

**Reply by thsubaku9** [reply]
Alright, then I guess this boils down to a homebrew error rather than the library one. Thanks for the quick update, I'll get in touch with the maintainers for this

**Reply by michaelklishin** [reply]
There is a lot of evidence that Erlang 24 works perfectly fine on Apple Silicon on Big Sur and the Homebrew formula is ready.
You should reinstall it without using Rosetta before filing any issues.

**Reply by thsubaku9** [reply]
It's working on direct m1, I think it was a rosetta translation problem. Thanks a lot :-)

**Comment by michaelklishin** [comment]
Also, `erlang@24` is [bottled for the Big Sur/Apple Silicon combination](https://github.com/Homebrew/homebrew-core/blob/master/Formula/erlang.rb#L15). In other words, a pre-compiled binary will be downloaded. There should be no reasons to use Rosetta.

**Comment by michaelklishin** [comment]
https://github.com/Homebrew/brew/issues/7857, https://github.com/erlang/otp/pull/2700, https://github.com/erlang/otp/pull/2687 are relevant Homebrew and Erlang/OTP issues I could find.
